process.go

```diff
@@ -48,26 +62,43 @@ func (p *process) Start(ctx context.Context) error {
 		p.io.Cancel()
 		p.io.Wait()
 		p.io.Close()
-		return err
+		return errdefs.FromGRPC(err)
 	}
+	p.mu.Lock()
 	p.pid = response.Pid
+	p.mu.Unlock()
 	return nil
 }
 
+// Kill sends the provided Signal to the process
 func (p *process) Kill(ctx context.Context, s syscall.Signal) error {
-	_, err := p.task.client.TaskService().Kill(ctx, &tasks.KillRequest{
+	if _, err := p.task.client.TaskService().Kill(ctx, &tasks.KillRequest{
 		Signal:      uint32(s),
 		ContainerID: p.task.id,
 		ExecID:      p.id,
-	})
-	return err
+	}); err != nil {
+		return errdefs.FromGRPC(err)
+	}
+	return nil
 }
 
+// Wiat is a blocking call that will wait for the process to exit and return the exit status
+//
+// Wait returns an ErrUnavailable when the task has already run and stopped
 func (p *process) Wait(ctx context.Context) (uint32, error) {
 	eventstream, err := p.task.client.EventService().Subscribe(ctx, &eventsapi.SubscribeRequest{})
 	if err != nil {
 		return UnknownExitStatus, err
 	}
+	// first check if the process has exited only if we have a pid
+	p.mu.Lock()
+	started := p.pid != 0
+	p.mu.Unlock()
+	if started {
+		if err := p.Kill(ctx, 0); err != nil {
+			return UnknownExitStatus, errdefs.ErrUnavailable
```

Comment:
Does kill not return the correct error code here? `Kill` should probably provide `errdefs.ErrUnavailable`. Perhaps, it would best to use `errors.Wrapf` here and provide some context.